src/argon2/_password_hasher.py

```diff
@@ -3,20 +3,34 @@
 from __future__ import annotations
 
 import os
+import platform
+import sys
 
 from typing import ClassVar, Literal
 
 from ._utils import Parameters, _check_types, extract_parameters
-from .exceptions import InvalidHashError
+from .exceptions import InvalidHashError, UnsupportedParamsError
 from .low_level import Type, hash_secret, verify_secret
 from .profiles import RFC_9106_LOW_MEMORY
 
 
+# this is a function because tests injects machine and platform
+# in PasswordHasher class. A global variable will be populated once and
+# we will need to import the file each time so that the mocking will be
+# effective. The function is used during initialization so it will not be
+# an overhead
+def is_wasm() -> bool:
```

Comment:
hi sorry for the late reaction.

I think I instead of checking for was and polluting the class, I would prefer to have a `validate_params_for_platform(params: Parameters) -> None` that raises the error.

This can be made efficient by having the `sys.platform` check at module level and defining different functions. For non-wasm it would be a `return None`. You can also set `DEFAULT_PARALLELISM` in the same block